### PR TITLE
[LWDM] feat(pay-card): page the card transaction history (LIVE-34780)

### PR DESCRIPTION
**Pull request by philipptpunkt**
### 📝 Description

The card transaction list read whatever single page `GET /v1/card/transactions` returned and stopped there. The provider pages that endpoint and offers no way to ask for everything, so a cardholder with more transactions than one page held could not reach the rest — and nothing on screen said so.

`getCardTransactions` becomes an RTK Query `infiniteQuery`. This PR is the data layer only: `useCardTransactionsViewModel` now exposes `loadMore` and `isLoadingMore`, and no surface calls them yet. The control that will lands in a follow-up on the history screen.

**Finding the end of the list is the hard part.** The provider answers a bare array — no total, no cursor, no `hasMore` — and a page *past* the end answers with **page 0 again** rather than an empty list, so overrunning looks like more data and the obvious "stop when a page comes back empty" never fires. No page size is documented either. `transactionsPaging.ts` therefore infers the end, in order of certainty:

1. an empty page;
2. a page holding **nothing but** transactions already read — the wraparound;
3. a page shorter than the widest one read so far, the widest standing in for the undocumented page size.

Rule 2 asks for a *whole* page of repeats rather than any, because a purchase landing between two reads shifts the page boundary down: a page that repeats only its first transaction is a genuine next page, and ending there would put every older transaction out of reach.

`getNextPageParam` only stops the *next* request, so the page that revealed a wraparound is already cached by then; `joinCardTransactionsPages` drops those repeats where the pages are joined.

Two smaller changes ride along:

- `page` moves off the filters onto the query's page param. `PayCardTransactionsRequest` is now filters only, and `PayCardTransactionsPageRequestSchema` validates the `{ queryArg, pageParam }` pair RTK Query actually hands to `argSchema` — validating the filters alone would reject every request.
- A page that fails **after** earlier ones landed no longer blanks the list. That fix sits in `resolveCardTransactionsDisplayState`, so it covers the Pay-tab preview and the history screen alike.

### 🔗 Context

- **JIRA / GitHub issue**: [LIVE-34780](https://ledgerhq.atlassian.net/browse/LIVE-34780)
- Provider reference: [Baanx — card transactions](https://docs.baanx.com/api-reference/card/transactions)

### 🧪 Testing

`transactionsPaging.test.ts` covers each rule, including the shifted-boundary case that rule 2 must *not* stop on. The view-model tests drive real paging through MSW, including the wraparound and a repeated press while a page is in flight.

Reviewers who want to see it end to end: the MSW mock pages (`pnpm desktop start:msw`, then DevTools → Card session → sign in with a mock session), though the visible control arrives in the follow-up.

### ⚠️ Known limits, called out deliberately

- **No page size is documented**, so a first page is never distinguishable from a full one. Nothing here guesses a size: assuming 20 when the truth is 10 would read a genuinely full page as the end and silently truncate a real history. The cost is one speculative request at the end of a list, whose duplicate rows are dropped.
- **A failed later page is silent** — the list stays, nothing announces the failure. Strictly better than the previous behaviour of blanking the section, but an inline error row is a design question, not a wiring one.
- The asset filter from #22231 filters client-side over this now-paged list, so with a filter active a fetched page may contain no matching rows and scrolling will fetch again. The endpoint filters by category, search key and date, but not by funding asset, so it cannot be pushed server-side.
- A `refetch()` on an infinite query replays every cached page in sequence, so a future pull-to-refresh over a deep history is N round trips. Nothing invalidates the `CardTransactions` tag today and no surface calls `refetch`, so it is latent.
- No page-count ceiling. If a new purchase landed before every overrun read, neither the repeat rule nor the short-page rule would fire; the walk is bounded only by how often the reader asks. `maxPages` would cap it if we ever want a belt.

<sub>Reviewed by the tracks pre-push reviewer: **pass**. The warnings it raised are the ones listed above, plus two that were fixed rather than carried: the joined pages are now sorted newest-first (a purchase landing mid-read could otherwise appear as the oldest row and open a duplicate day header), and a test name that credited the wrong mechanism.</sub>

[LIVE-34780]: https://ledgerhq.atlassian.net/browse/LIVE-34780?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: domain/api/card-management/src/transactionsPaging.ts [hunk 1]
```diff
@@ -0,0 +1,71 @@
...
+  if (lastPage.length === 0) {
+    return undefined;
+  }
+
+  // A page holding nothing but transactions already read is the wraparound: page 0 served again.
+  // Every transaction, not any: a purchase landing between two reads shifts the page boundary down,
+  // so a page that repeats only its first transaction is a genuine next page, not the end.
+  const alreadyRead = new Set(allPages.slice(0, -1).flatMap(page => page.map(({ id }) => id)));
+  if (lastPage.every(({ id }) => alreadyRead.has(id))) {
+    return undefined;
```
[the remark this answers is not in the document]
Comment (philipptpunkt) [reply domain/api/card-management/src/transactionsPaging.ts]: valid concern in the longer run, but the solution is a real pagination on BE side, rather than more hacks on our side...

**Review by mcayuelas-ledger — DISMISSED** [review]

**Review by liviuciulinaru — APPROVED** [review]

**Review by liviuciulinaru — DISMISSED** [review]

File: domain/api/card-management/src/cardTransactions.mock.ts [hunk 2]
```diff
@@ -231,3 +231,24 @@ export function receivePayCardTransactionMock(asset: PayCardMockTransactionAsset
...
+ * One page of the mocked history, answering as the provider does: a numeric page past the end
+ * gives an empty array, while a missing or non-numeric one falls back to page 0.
+ */
+export function mockPayCardTransactionsPage(request: Request) {
+  const all = readPayCardTransactionsMock() ?? mockPayCardTransactions();
+  const requested = Number(new URL(request.url).searchParams.get("page"));
+  const page = Number.isInteger(requested) && requested >= 0 ? requested : 0;
+  const start = page * MOCK_CARD_TRANSACTIONS_PAGE_SIZE;
+
+  return all.slice(start, start + MOCK_CARD_TRANSACTIONS_PAGE_SIZE);
```
[the remark this answers is not in the document]
Comment (philipptpunkt) [reply domain/api/card-management/src/cardTransactions.mock.ts:253]: Based on example from the API docs and confirmed by Baanx

File: domain/api/card-management/src/transactionsPaging.ts [hunk 3]
```diff
@@ -0,0 +1,47 @@
...
+ *
+ * An empty page is the end. A history that exactly fills its last page therefore costs one further
+ * request, which comes back empty.
+ */
+export function nextCardTransactionsPage(
+  lastPage: readonly PayCardTransaction[],
+  _allPages: readonly (readonly PayCardTransaction[])[],
+  lastPageParam: number,
+): number | undefined {
+  return lastPage.length === 0 ? undefined : lastPageParam + 1;
```
[the remark this answers is not in the document]
Comment (philipptpunkt) [reply domain/api/card-management/src/transactionsPaging.ts:17]: The page-0 replay only applies when `page` is missing or not a number — Baanx confirmed this directly and are correcting the spec wording that says otherwise — and we always send a schema-validated non-negative integer, so that branch is unreachable here and an empty page is a reliable terminator. We will re-evaluate whether a termination guard is still worth having once we have tested against real data.

**Review by liviuciulinaru — DISMISSED** [review]

**Review by tonykhaov — DISMISSED** [review]

**Review by liviuciulinaru — APPROVED** [review]

File: features/flow/pay-card-widget/src/onboardingStatus/useCardOnboardingSources.ts [hunk 4]
```diff
@@ -55,7 +55,11 @@ export function useCardOnboardingSources({
       // and reading it as "no card" would send the holder back to choosing a type.
       "choose-card-type": cardStatus.data !== undefined,
       "top-up-card": linkedWallets.wallets.some(({ balance }) => hasPositiveBalance(balance)),
-      "first-purchase": transactions.data?.some(({ status }) => status === "CONFIRMED") === true,
+      // Every page read, not just the first: the newest page can hold nothing but pending or
+      // declined attempts while an older charge did settle.
+      "first-purchase": (transactions.data?.pages ?? []).some(page =>
+        page.some(({ status }) => status === "CONFIRMED"),
+      ),
```
[the remark this answers is not in the document]
Comment (philipptpunkt) [reply features/flow/pay-card-widget/src/onboardingStatus/useCardOnboardingSources.ts:62]: 'hasConfirmedCardTransactions' was reverted

**Review by tonykhaov — APPROVED** [review]
